resolver/restriction_resolver.go

```diff
@@ -0,0 +1,100 @@
+package resolver
```

Comment:
This filename should be cname_resolver, since that equates to what it actually does, while we (currently) are only using it only for applying restrictions.